File: lib/stack.rb
# frozen_string_literal: true

# A queue class
class Stack
  attr_reader :store
  private :store

  def initialize
    @store = []
  end

  def empty?
    store.empty?
  end

  def push(value)
    store.push(value)
    self
  end

  def pop
    store.pop
  end
end

File: lib/stack.js
'use strict';

const Stack = function () {
  Object.defineProperty(this, '_store', {
    value: [],
  });
};

Object.defineProperty(Stack.prototype, 'isEmpty', {
  get() {
    return this._store.length === 0;
  },
});

Stack.prototype.push = function(value) {
  this._store.push(value);
  return this;
};

Stack.prototype.pop = function() {
  return this._store.pop();
};

module.exports = Stack;
